src/utils/hooks/useSingleProduct.js

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from 'react';
+import { API_BASE_URL } from '../constants';
+import { useLatestAPI } from './useLatestAPI';
+
+export function useSingleProduct(productId) {
+  const { ref: apiRef, isLoading: isApiMetadataLoading } = useLatestAPI();
+  const [product, setProduct] = useState(() => ({
+    data: {},
+    isLoading: true,
+  }));
+
+  useEffect(() => {
+    if (!apiRef || isApiMetadataLoading) {
+      return () => {};
+    }
+
+    const controller = new AbortController();
+
+    async function getProduct() {
+      try {
+        setProduct({ data: {}, isLoading: true });
+
+        const response = await fetch(
+          `${API_BASE_URL}/documents/search?ref=${apiRef}&q=${encodeURIComponent(
+            '[[at(document.type, "product")]]'
+          )}&q=${encodeURIComponent(`[[at(document.id, "${productId}")]]`)}&lang=en-us&pageSize=1`,
+          {
+            signal: controller.signal,
+          }
+        );
+        const data = await response.json();
+
+        setProduct({ data, isLoading: false });
+      } catch (err) {
+        setProduct({ data: {}, isLoading: false });
+        console.error(err);
+      }
+    }
+
+    getProduct();
```

Comment:
all the hook that fetch from the API have duplicated code, the only thing that changes is the params passed to the request, a new hook called `useFetch` or something could be used to abstract the log to fetch the API and have the pother hooks pass the params to that hook, it would make the code maintainable